### PR TITLE
feat: Add ability to update a person

### PR DESCRIPTION
**Pull request by dweebo**
First of all thanks for building this library!  

I've been tasked with updating some of our activists records to set some custom fields based on BOE data and this library is going to make the task MUCH less tedious. 

I needed a way to update a persons records so this PR adds that.

See updated examples in the README.

Thanks again!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,8 @@ Or install it yourself as:
 ## Usage
 
 ```
+require 'action_network_rest'
```
Comment (woodhull) [inline README.md:26]: 💯

File: lib/action_network_rest/people.rb [hunk 2]
```diff
@@ -41,6 +41,12 @@ def find_by_email(email)
       end
     end
 
+    def update(person)
```
Comment (woodhull) [inline lib/action_network_rest/people.rb]: It seems like potentially we should also support a hash and an ID as input to be more flexible re the sort of data we are accepting if the consuming code does.not have a person object handy.

Is it possible to send an update with a single field in a hash to not overwrite the entire person with new data for eg?

**Review by woodhull — COMMENTED** [review]
Thanks for this. Would prefer to keep the interface for create/update a hash and ID like the other similar methods in the gem for other resources. It's probably more likely that consuming code has those simpler data types than a struct like object that our GETs return.

If you make that tweak would be happy to merge and cut a new version of gem.

**Comment by dweebo** [comment]
> Thanks for this. Would prefer to keep the interface for create/update a hash and ID like the other similar methods in the gem for other resources. It's probably more likely that consuming code has those simpler data types than a struct like object that our GETs return.
> 
> If you make that tweak would be happy to merge and cut a new version of gem.

I was wondering about that, I did see that update pattern in a few other spots.  

For my purposes I will need to first lookup a user by email using your existing code, so I do already have the person object but I see what you are saying.

I'll rework this PR

**Comment by dweebo** [comment]
Updated as requested to pass in hash and ID like other `update` methods.  Let me know if you see anything else needed.

**Pull request merged**